### PR TITLE
Add SwiftSyntax 509.1.1

### PR DESCRIPTION
**Pull request by SimplyDanny**
https://github.com/apple/swift-syntax/releases/tag/509.1.1

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @keith, modules you maintain (swift-syntax) have been updated in this PR. Please review the changes.

**Comment by SimplyDanny** [comment]
I assume [apple/swift-syntax](https://github.com/apple/swift-syntax) needs to provide a stable source code archive as part of their release first ...

**Comment by meteorcloudy** [comment]
Can you please add bazel versions to the presubmit.yml file? 
Context: https://github.com/bazelbuild/bazel-central-registry/pull/1387

File: modules/swift-syntax/509.1.1/presubmit.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+tasks:
+  verify_targets:
+    name: Verify build targets
+    platform: macos
```
Comment (keith) [inline modules/swift-syntax/509.1.1/presubmit.yml:4]:
```suggestion
    platform: macos
    bazel: 7.x
```

File: modules/swift-syntax/509.1.1/presubmit.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    bazel: 7.x
+    build_targets:
+      - '@swift-syntax//...'
+    build_flags:
+      - --macos_minimum_os=12.0
```
Comment (keith) [inline modules/swift-syntax/509.1.1/presubmit.yml:9]:
```suggestion
      - --macos_minimum_os=12.0
      - "--repo_env=BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=1"
```

**Review by keith — APPROVED** [review]

**Pull request merged**